mesa_frames/abstract/space.py

```diff
@@ -321,3 +383,345 @@ def random(self) -> Generator:
         Generator
         """
         return self.model.random
+
+
+class DiscreteSpaceDF(SpaceDF):
+    """The DiscreteSpaceDF class is an abstract class that defines the interface for all discrete space classes (Grids and Networks) in mesa_frames.
+
+    Methods
+    -------
+    __init__(model: 'ModelDF', capacity: int | None = None)
+        Create a new DiscreteSpaceDF object.
+    is_free(pos: DiscreteCoordinate | DiscreteCoordinates) -> DataFrame
+        Check whether the input positions are free (there exists at least one remaining spot in the cells).
+    is_empty(pos: DiscreteCoordinate | DiscreteCoordinates) -> DataFrame
+        Check whether the input positions are empty (there isn't any single agent in the cells).
+    is_full(pos: DiscreteCoordinate | DiscreteCoordinates) -> DataFrame
+        Check whether the input positions are full (there isn't any spot available in the cells).
+    move_to_empty(agents: IdsLike | AgentContainer | Collection[AgentContainer], inplace: bool = True) -> Self
+        Move agents to empty cells in the space (cells where there isn't any single agent).
+    move_to_free(agents: IdsLike | AgentContainer | Collection[AgentContainer], inplace: bool = True) -> Self
```

Comment:
:+1: yes this is a helpful method for multi-grid. Thanks for innovating in the API.

Reply:
It's a bit longer, but I think `move_to_available` is clearer, do you agree? The opposite of free table is still occupied table, even though the table can have multiple seats left.

Reply:
You could also say available table <-opposite-> occupied table. Huh surprising that there is no easy word/phrase to say this.

Reply:
Yes "available" is clearer. I couldn't think of that term at the time ahah. I would leave "full" since it is short and clear enough